docs/reference/modules/piece-page-type.md

```diff
@@ -0,0 +1,240 @@
+
+This option can also be set to an object containing an object with a key of `project` and values which will be used as a query builder for retrieving the previous piece document.
+
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+  extend: '@apostrophecms/piece-page-type',
+  options: {
+    previous: true
+  },
+  // …
+}
+
+// OR
+
+module.exports = {
+  extend: '@apostrophecms/piece-page-type',
+  options: {
+    // The previous article piece would be returned with only the `title`, `_url`
+    // and `lastPublishedAt` properties.
+    next: {
+      project: {
+        title: 1,
+        _url: 1,
+        lastPublishedAt: 1
+      }
+    }
+  },
+  // …
+}
+```
+<template v-slot:caption>
+  modules/article-page/index.js
+</template>
+</AposCodeBlock>
+
+## Related Documentation
+
+* [Piece index and show pages](https://v3.docs.apostrophecms.org/guide/piece-pages.html)
+* [Piece page type options](https://v3.docs.apostrophecms.org/reference/module-api/module-options.html#options-for-piece-page-types)
+
+## Featured methods
+The following methods belong to this module and may be useful in project-level code. See the [source code](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/piece-page-type/index.js) for all the modules that belong to this module.
+
+### `indexQuery(req)`
+This method should be overridden for a piece-type to call additional [query builders](https://v3.docs.apostrophecms.org/reference/query-builders.html#query-builders) by default.
+
+### `beforeIndex(req)`
+This method is called before `indexPage`. It is a convenient method to extend for manipulating the `req` being supplied to that page.
+
+### `beforeShow(req)`
+This method is called before `showPage`. It is a convenient method to extend for manipulating the `req` being supplied to that page.
+
+### `dispatchAll()`
+This method can be extended to override the default behavior of invoking `showPage` if the URL has an additional path after the base, e.g. `/blog/good-article`. As example, you could override to use `/:year/:month/:day/:slug` to invoke `self.showPage`. This should be used in conjunction with the [`buildUrl()`](#buildurl-req-page-piece)) method of this module. See [@apostrophecms/page-type](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/page-type/index.js) for more about what you can do with dispatch routes.
+
+### `buildUrl(req, page, piece)`
+This method should be extended to build custom URL for use in the [`dispatchAll`](#dispatchall) method.
+
+### `filterByIndexPage(query, page)`
+This method invokes query builders on the supplied query argument to ensure it only fetches results appropriate to the given page. This is typically done when there is more than one pieces-page per page type. This should be used in conjunction with the [`chooseParentPage`](#chooseparentpage-pages-piece) method of this module.
```

Comment:
Please make clear that this method does nothing by default, and should be overridden to have the effect described in use cases where it is needed. Also note this clearly for other methods mentioned here that are empty in core.

Overriding and extending are notably different in Apostrophe because the easiest way to override (completely replace) a method is to just write it in your own module's `methods` section, while the easiest way to extend a method (invoke the original, but return a modified result, etc) is to use the module's `extendMethods` section. So we should use the word "override" specifically when a method is initially empty by design and you don't need to worry about using `extendMethods`.

Reply:
Changed for `filterByIndexPage`, `beforeIndex`, and `beforeShow`.